lib/internal/url.js

```diff
@@ -442,7 +445,9 @@ class URLSearchParams {
       }
     }
 
-    update(this[context], this);
+    if (this[context]) {
+      this[context].search = this.toString();
```

Comment:
How does this work? `update` also updated `this[context].path` and `this[context].href`, but this doesn't seem to do that.

Reply:
Thanks for asking!

`this[context]` refers to URL which directly calls the setter of `search` which triggers the callback of updating href etc. Previously, we were doing this in the javascript layer, but it was creating the reasoning of having the business logic of URL inside Node.js core. I am working on improving each setter performance, and before doing that I wanted to remove all the logic from JavaScript layer.